js/src/admin/utils/ExtensionData.js

```diff
@@ -0,0 +1,34 @@
+import ItemList from '../../common/utils/ItemList';
+import listItems from '../../common/helpers/listItems';
+
+export default class ExtensionData {
```

Comment:
As per some of my previous comments, I think this should work in tandem with a class that encapsulates storage (and retrieval) of all these pages, permissions, and settings.